Framework/API/src/DetectorInfo.cpp

```diff
@@ -315,6 +312,27 @@ const Geometry::IComponent &DetectorInfo::getSample() const {
   return *m_sample;
 }
 
+/** Returns a reference to a vector of detector indices in a CompAssembly.
+ *
+ * Note that this method will be removed once ComponentInfo provides this
+ * functionality. */
+const std::vector<size_t> &DetectorInfo::getAssemblyDetectorIndices(
+    const Geometry::IComponent &comp) const {
+  const auto *base = comp.getBaseComponent();
+  size_t thread = static_cast<size_t>(PARALLEL_THREAD_NUMBER);
+  if (m_lastAssemblyDetectorIndices[thread].first != base) {
+    m_lastAssemblyDetectorIndices[thread].first = base;
+    m_lastAssemblyDetectorIndices[thread].second.clear();
+    std::vector<Geometry::IDetector_const_sptr> dets;
```

Comment:
Once you have the detector vector you could do `m_lastAssemblyDetectorIndices[thread].second.reserve(dets.size())` - some of the assemblies could be quite big.